packages/running/src/index.tsx

```diff
@@ -69,140 +64,89 @@ const ITEM_LABEL_CLASS = 'jp-RunningSessions-itemLabel';
  */
 const SHUTDOWN_BUTTON_CLASS = 'jp-RunningSessions-itemShutdown';
 
+/* tslint:disable */
 /**
- * The class name added to a notebook icon.
- */
-const NOTEBOOK_ICON_CLASS = 'jp-mod-notebook';
-
-/**
- * The class name added to a console icon.
- */
-const CONSOLE_ICON_CLASS = 'jp-mod-console';
-
-/**
- * The class name added to a file icon.
+ * The running sessions token.
  */
-const FILE_ICON_CLASS = 'jp-mod-file';
+export const IRunningSessionManagers = new Token<IRunningSessionManagers>(
+  '@jupyterlab/running:IRunningSessionManagers'
+);
+/* tslint:enable */
 
 /**
- * The class name added to a terminal icon.
+ * The running interface.
  */
-const TERMINAL_ICON_CLASS = 'jp-mod-terminal';
-
-/**
- * Properties for a session list displaying items of generic type `M`.
- */
-type SessionProps<M> = {
-  /**
-   * A signal that tracks when the `open` is clicked on a session item.
-   */
-  openRequested: Signal<RunningSessions, M>;
-
-  /**
-   * The session manager.
-   */
-  manager: {
-    /**
-     * The function called when the shutdown all button is pressed.
-     */
-    shutdownAll(): void;
-
-    /**
-     * A signal that should emit a new list of items whenever they are changed.
-     */
-    runningChanged: ISignal<any, M[]>;
-
-    /**
-     * Returns a list the running models.
-     */
-    running(): IIterator<M>;
-  };
-
+export interface IRunningSessionManagers {
   /**
-   * The function called when the shutdown button is pressed on an item.
+   * Add a running item manager.
+   *
+   * @param manager - The running item manager.
+   *
    */
-  shutdown: (model: M) => void;
-
-  /**
-   * The filter that is applied to the items from `runningChanged`.
-   */
-  filterRunning?: (model: M) => boolean;
-
-  /**
-   * The name displayed to the user.
-   */
-  name: string;
-
+  add(manager: IRunningSessions.IManager): void;
```

Comment:
How do we remove things we've added? One common pattern we have in the code is to return a disposable, and when you dispose it, it removes the item from the registry. A simpler way might be to just have a remove method.

Reply:
Disposable sounds like the right way, if not the easy way, can you point me at an example?

Reply:
Although its possible that the intention was that this would be add-only